x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -0,0 +1,208 @@
+    );
+
+    // get and save pipeline refs before installing pipelines
+    // eslint-disable-next-line no-console
+    console.log(`transform: ${JSON.stringify(transformRefs)}`);
+    await saveInstalledEsRefs(savedObjectsClient, registryPackage.name, transformRefs);
+
+    const transforms: TransformInstallation[] = transformPathDatasets.map(
+      (transformPathDataset: TransformPathDataset) => {
+        return {
+          installationName: getTransformNameForInstallation(
+            transformPathDataset,
+            installNameSuffix
+          ),
+          content: Registry.getAsset(transformPathDataset.path).toString('utf-8'),
+        };
+      }
+    );
+
+    const installationPromises = transforms.map(async (transform) => {
+      return installTransform({ callCluster, transform });
+    });
+
+    const installedTransforms = await Promise.all([
+      Promise.all(installationPromises),
+    ]).then((results) => results.flat());
+
+    const currentInstallation = await getInstallation({
+      savedObjectsClient,
+      pkgName: registryPackage.name,
+    });
+
+    // delete all previous transform
+    await deleteTransforms(
+      callCluster,
+      previousInstalledTransformEsAssets.map((asset) => asset.id)
+    );
+
+    // remove the saved object reference
+    await deleteTransformRefs(
+      savedObjectsClient,
+      currentInstallation?.installed_es || [],
+      registryPackage.name,
+      previousInstalledTransformEsAssets.map((asset) => asset.id)
+    );
+
+    return installedTransforms;
+  } catch (e) {
+    // delete all resources we tried to install
+    // restart old transform again
+    throw e;
+  }
+};
+
+const timestamp = () => {
+  const dt = new Date();
+  return `${dt.getFullYear()}${dt.getUTCMonth()}${dt.getUTCDay()}${dt.getUTCMinutes()}`;
+};
+
+const isTransform = (path: string) => {
+  const pathParts = Registry.pathParts(path);
+  return pathParts.type === ElasticsearchAssetType.transform;
+};
+
+const isDatasetTransform = (path: string, datasetName: string) => {
+  const pathParts = Registry.pathParts(path);
+  return (
+    !path.endsWith('/') &&
+    pathParts.type === ElasticsearchAssetType.transform &&
+    pathParts.dataset !== undefined &&
+    datasetName === pathParts.dataset
+  );
+};
+
+async function installTransform({
+  callCluster,
+  transform,
+}: {
+  callCluster: CallESAsCurrentUser;
+  transform: TransformInstallation;
+}): Promise<EsAssetReference> {
+  const callClusterParams: {
+    method: string;
+    path: string;
+    query: string;
+    ignore?: number[];
+    body: any;
+    headers?: any;
+  } = {
+    method: 'PUT',
+    path: `/_transform/${transform.installationName}`,
+    query: 'defer_validation=true',
+    body: transform.content,
+  };
+  /* if (pipeline.extension === 'yml') {
+      callClusterParams.headers = { ['Content-Type']: 'application/yaml' };
+    }*/
+
+  // This uses the catch-all endpoint 'transport.request' because we have to explicitly
```

Comment:
Comment is probably copied over but I wonder if we support yaml transforms or not?

@skh @neptunian It would be nice to have a general solution for yaml vs json that always both are supported.

Reply:
In the pipeline installation this is not commented out:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/ingest_pipeline/install.ts#L158-L160

> I wonder if we support yaml transforms or not?

As this is the initial implementation of transform support, isn't the question whether we *want* to support yaml transforms or not?

Reply:
@skh You are correct, we don't need it right now. What I wonder about is we could find a solution that we don't need to have this discussions for each assets for Elasticsearch but have a more generic solution. I'll open a separate issue for it.

Reply:
Hence I commented it out. I was trying one case out. We can add it in phases.